File: 第五章_链表/链表(2)/链表指定值清除练习题/ClearValue.java
public class ListNode {
	int val;
	ListNode next = null;

	ListNode(int val) {
		this.val = val;
	}
}

public class ClearValue {
	public ListNode clear(ListNode head, int val) {
		ListNode pListNode = head;
		ListNode node = new ListNode(0);// 创建一个指向head的指针
		node.next = pListNode;
		ListNode cur = node;
		while (pListNode != null) {// 遍历链表 删除目标值
			if (pListNode.val == val) {
				node.next = pListNode.next;
			} else {
				node = node.next;
			}
			pListNode = pListNode.next;
		}

		return cur.next;
	}
}

File: 第五章_链表/链表(2)/链表指定值清除练习题/ClearValue.cpp
/*
struct ListNode {
    int val;
    struct ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};*/
class ClearValue {
public:
    ListNode* clear(ListNode* head, int val) {
        
       ListNode *node=new ListNode(0);// 创建一个指向head的指针
       node->next=head;
       ListNode *p=node;
        ListNode *cur=head;
       while(cur!=NULL)// 遍历链表 删除目标值
       {
           if(cur->val==val)
           {
              p->next=cur->next;
           }
           else
           {
               p=p->next;
           }
           cur=cur->next;
       }
 
       return node->next;
        
    }
}